lib/passport-saml/saml.js

```diff
@@ -661,9 +661,14 @@ SAML.prototype.validatePostResponse = function (container, callback) {
   })
   .then(() => this.certsToCheck())
   .then(certs => {
-    // Check if this document has a valid top-level signature
+    // Check if this document has a valid top-level signature which applies to the entire XML document
     var validSignature = false;
-    if (this.options.cert && this.validateSignature(xml, doc.documentElement, certs)) {
+    if (this.options.cert &&
+      this.validateSignature(xml, doc.documentElement, certs) &&
+      Array.from(doc.childNodes).filter(
+          (n) => n.tagName != null && n.childNodes != null
+        ).length === 1
+    ) {
       validSignature = true;
```

Comment:
Released in passport-saml library 3.2.2 https://github.com/node-saml/passport-saml/commit/8b7e3f5a91c8e5ac7e890a0c90bc7491ce33155e